paasta_tools/cli/cmds/mark_for_deployment.py

```diff
@@ -1323,143 +1323,88 @@ def check_if_instance_is_done(
             )
             return False
 
-    log.debug(
-        "Inspecting the deployment status of {}.{} on {}".format(
-            service, instance, cluster
-        )
-    )
+    inst_str = f"{service}.{instance} in {cluster}"
+    log.debug(f"Inspecting the deployment status of {inst_str}")
+
+    status = None
     try:
-        status = None
-        status = api.service.status_instance(
-            service=service,
-            instance=instance,
-            include_smartstack=False,
-            include_envoy=False,
-            include_mesos=False,
-        )
+        status = api.service.bounce_status_instance(service=service, instance=instance)
     except api.api_error as e:
-        if e.status == 404:
+        if e.status == 404:  # non-existent instance
             # TODO(PAASTA-17290): just print the error message so that we
             # can distinguish between sources of 404s
             log.warning(
                 "Can't get status for instance {}, service {} in "
                 "cluster {}. This is normally because it is a new "
-                "service that hasn't been deployed by PaaSTA yet".format(
+                "service that hasn't been deployed by PaaSTA yet.".format(
                     instance, service, cluster
                 )
             )
-        else:
+        else:  # 500 - error talking to api
             log.warning(
                 "Error getting service status from PaaSTA API for "
                 f"{cluster}: {e.status} {e.reason}"
             )
 
-    long_running_status = None
-    if status:
-        if status.marathon:
-            long_running_status = status.marathon
-        if status.kubernetes:
-            long_running_status = status.kubernetes
-    if not status:
-        log.debug(
-            "No status for {}.{}, in {}. Not deployed yet.".format(
-                service, instance, cluster
-            )
-        )
+        log.debug(f"No status for {inst_str}. Not deployed yet.")
         return False
-    elif not long_running_status:
-        log.debug(
-            "{}.{} in {} is not a Marathon or Kubernetes job. Marked as deployed.".format(
-                service, instance, cluster
-            )
-        )
-        return True
-    elif (
-        long_running_status.expected_instance_count == 0
-        or long_running_status.desired_state == "stop"
-    ):
+
+    if not status:  # 204 - instance is not bounceable
```

Comment:
I tested this by forcing a locally running paasta api to return a 204 immediately (though the bounce status endpoint does naturally return one), and status comes back as an empty string. It is likely because the way we construct the 204 response is create one using `Response()`, setting the status code to 204, then returning it immediately.